### ISSUE TITLE
Division of 12

### ISSUE THREAD

**Issue by Rangoric** [Em3rgencyLT/Hexasphere#2]
I haven't had a chance to delve too deep into the issue, but I got an error of:

```
InvalidOperationException: Sequence contains no matching element
System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) (at <b98680caec7f4227a10f36469340208c>:0)
Code.Hexasphere.Point.GetOrderedFaces () (at Assets/Scripts/Hexasphere/Point.cs:80)
Code.Hexasphere.Tile..ctor (Code.Hexasphere.Point center, System.Single radius, System.Single size) (at Assets/Scripts/Hexasphere/Tile.cs:29)
Code.Hexasphere.Hexasphere.<ConstructTiles>b__17_0 (Code.Hexasphere.Point point) (at Assets/Scripts/Hexasphere/Hexasphere.cs:143)
System.Collections.Generic.List`1[T].ForEach (System.Action`1[T] action) (at <7b935204f5ff4bcab44b3b0ebca330bf>:0)
Code.Hexasphere.Hexasphere.ConstructTiles () (at Assets/Scripts/Hexasphere/Hexasphere.cs:141)
Code.Hexasphere.Hexasphere..ctor (System.Single radius, System.Int32 divisions, System.Single hexSize) (at Assets/Scripts/Hexasphere/Hexasphere.cs:29)
```

While running the default scripts you have with divisions = 12.

From what I saw so far, it is the ordered hexagon code, where you pick the next neighboring hexagon, but it can't find any, but there are more left.

Values up to 16 work fine otherwise from my basic testing. Just 12 seems to have an issue.

**Comment by Em3rgencyLT** [comment]
This was happening because 2 of the faces should have shared a point, but the subdivision put them apart by 0.0000001f units. Coincidentally, I had set the rounding error for when comparing if two points are the same or not to 0.000001f. Thus 2 faces that should be sharing a vertex instead had 2 separate vertices almost at the same location and thus weren't considered neighbors anymore.

I've reduced the rounding error to 0.0001f as a fix.

This issue will repeat if really small spheres are made, IE less than 1 unit radius. It should never happen on properly sized spheres though.
